src/utils/parseHTML.ts

```diff
@@ -38,13 +38,15 @@ const parseHTML = (html: string): ReactElement | ReactElement[] | string => {
                 if (name.match(/^h[0-9]$/)) {
                     return rewriteHeader(domNode);
                 }
+                if (name === 'callout') {
+                    return parseCalloutTags(domNode.attribs.type, domToReact(domNode.children));
+                }
```

Comment:
We don't need the `&& domNode.parent?.name !== 'code'` protection anymore, because code blocks won't have rendered the p tag into a callout tag.